packages/synapse-client/out/modelGeneric.mustache

```diff
@@ -0,0 +1,168 @@
+import { mapValues } from '../runtime{{importFileExtension}}';
+{{#hasImports}}
+{{#tsImports}}
+import type { {{{classname}}} } from './{{filename}}{{importFileExtension}}';
+import {
+    {{classname}}FromJSON,
+    {{classname}}FromJSONTyped,
+    {{classname}}ToJSON,
+} from './{{filename}}{{importFileExtension}}';
+{{/tsImports}}
+
+{{/hasImports}}
+{{#discriminator}}
+{{#discriminator.mappedModels}}
+import { {{modelName}}FromJSONTyped } from './{{modelName}}{{importFileExtension}}';
+{{/discriminator.mappedModels}}
+{{/discriminator}}
+{{>modelGenericInterfaces}}
+
+/**
+ * Check if a given object implements the {{classname}} interface.
+ */
+export function instanceOf{{classname}}(value: object): value is {{classname}} {
+    {{#vars}}
+    {{#required}}
+    if (!('{{name}}' in value) || value['{{name}}'] === undefined) return false;
+    {{! START CUSTOM TEMPLATING FOR synapse-client }}
+    {{! if the property is a required enum, check that the value is a valid enum value }}
+    {{#isEnum}}
+      if (!Object.values({{classname}}{{enumName}}).includes(value["{{name}}"] as {{classname}}{{enumName}})) {
+        return false
+      }
+    {{/isEnum}}
+    {{! END CUSTOM TEMPLATING FOR synapse-client }}
```

Comment:
In the `instanceOf` method, if a model's property is a required enum, but the property's value does not match a known enum value, return `false`. This is especially useful because it will use our `concreteType` property, which is now always defined as an enum following PLFM-8257

Here is an example of generated code that includes this logic. See that the generated code checks that the `concreteType` matches one of the allowed enumeration values, which in this case is only `'org.sagebionetworks.repo.model.FileEntity'`.

```typescript
/**
 * Check if a given object implements the FileEntity interface.
 */
export function instanceOfFileEntity(value: object): value is FileEntity {
  if (!('concreteType' in value) || value['concreteType'] === undefined)
    return false
  if (
    !Object.values(FileEntityConcreteTypeEnum).includes(
      value['concreteType'] as FileEntityConcreteTypeEnum,
    )
  ) {
    return false
  }
  return true
}
```

Reply:
Oh I see!  Thanks for the example output, very clever